desktop/src/com/frostwire/gui/bittorrent/BTDownloadMediator.java

```diff
@@ -674,7 +672,11 @@ public void run() {
                     }
 
                     if (saveDir == null) {
-                        BTEngine.getInstance().download(torrentFile, null, filesSelection);
+                        if (ConnectionSettings.MANDATORY_VPN_FOR_BITTORRENT.getValue() && !VPNs.isVPNActive()) {
```

Comment:
I guess this check should be moved before this `if (saveDir == null)` altogether (and add a `return`), otherwise the `openTorrentForSeed` case will occur.

I guess I'd move it all the way to the start of the method as a short-circuit mechanism.